sympy/matrices/random.py

```diff
@@ -0,0 +1,2292 @@
+from random import Random
+
+from ..core import I as _i
+from ..core.mul import Mul as _multiply
+from ..functions import sqrt as _sqrt, re as _re, im as _im, \
+    transpose as _t, adjoint as _c
+from ..tensor import shape as _shape
+from .dense import eye as _eye
+
+__all__ = 'projection', 'jordan', 'transposition', \
+          'permutation', 'elementary', 'rotation', 'reflection', \
+          'diagonal_normal', 'jordan_normal', 'isometry_normal', \
+          'triangular', 'invertible', 'singular', \
+          'idempotent', 'nilpotent', 'diagonalizable', 'trigonalizable', \
+          'orthogonal', 'unitary', 'normal', \
+          'symmetric', 'hermite', \
+          'regular_to_singular', 'complex_to_real', 'rand'
+
+""" random generator """
+rand = Random()
```

Comment:
Is this thread-safe?

Reply:
I did search for a common interface but found more than on. Finally, I switched to a flexible but hopefully intuitiv version https://github.com/sympy/sympy/pull/22103#discussion_r711015017